### PR TITLE
Jomah Fangonilo

### PR DESCRIPTION
**Pull request by jfangonilo**

### CODE REVIEW

File: lib/house.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    @address = address
+    @rooms = []
+  end
+
+  def add_room(room)
+    @rooms << room
+  end
+
+  def rooms_from_category(category)
+    @rooms.find_all { |room| room if room.category == category }
```
Comment (aziobrow) [inline lib/house.rb]: nice use of the appropriate tool for this job (i.e., choosing find_all instead of each)

File: lib/house.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  end
+
+  def area
+    @rooms.inject(0) { |sum, room| sum + room.area }
```
Comment (aziobrow) [inline lib/house.rb]: YES! Love this-- perfect use case for inject (also called reduce, which is probably the term more prevalently used in other languages).

File: lib/house.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  end
+
+  def price_per_sqare_foot
+    (price_as_float / area).round(2)
+  end
+
+  def price_as_float
```
Comment (aziobrow) [inline lib/house.rb]: I like the concept here, but pay attention to the interaction pattern: 

```
pry(main)> house.price
#=> 400000
```

price should already be an integer (not a float) without a dollar sign, so that's a clue that we shouldn't be doing this work in a method (unless it's a method called price)

File: lib/room.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+class Room
+  attr_reader :category, :length, :width, :area
+
+  def initialize(category, length, width)
+    @category = category
+    @length = length
+    @width = width
+    @area = length * width
```
Comment (aziobrow) [inline lib/room.rb]: In general, we try to avoid doing work (such as multiplying) in initialize.  Remember, attributes are *state*, not *behavior*, so area is more appropriate as a method here. Doesn't hurt anything in particular, but definitely best practice in this case.

File: test/house_test.rb [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    @room_4 = Room.new(:basement, 30, 41) #1230
+    @house = House.new("$400000", "123 sugar lane")
+  end
+
+  def test_it_exists
+    assert_instance_of House, @house
+  end
+
+  def test_it_has_a_price
+    assert_equal "$400000", @house.price
```
Comment (aziobrow) [inline test/house_test.rb]: Again, callout that the interaction pattern shows that this is an integer with no dollar sign

File: test/house_test.rb [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+  def test_it_has_an_address
+    assert_equal "123 sugar lane", @house.address
+  end
+
+  def test_add_room
+    assert_equal [], @house.rooms
+    @house.add_room(@room_1)
+    assert_equal [@room_1], @house.rooms
+    @house.add_room(@room_2)
+    assert_equal [@room_1, @room_2], @house.rooms
```
Comment (aziobrow) [inline test/house_test.rb]: nice progressive testing here

File: test/house_test.rb [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+  end
+
+  def test_rooms_from_category
+    @house.add_room(@room_1)
+    @house.add_room(@room_2)
+    @house.add_room(@room_3)
+    @house.add_room(@room_4)
+
+    assert_equal [@room_1, @room_2], @house.rooms_from_category(:bedroom)
+    assert_equal [@room_4], @house.rooms_from_category(:basement)
```
Comment (aziobrow) [inline test/house_test.rb]: nice. what happens if you pass in a category that doesn't exist? always a good edge case test for something like this

File: test/house_test.rb [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+  end
+
+  def test_rooms_by_category
+    @house.add_room(@room_1)
+    @house.add_room(@room_2)
+    @house.add_room(@room_3)
+    @house.add_room(@room_4)
+
+    sorted = {bedroom: [@room_1, @room_2], living_room: [@room_3], basement: [@room_4]}
+    assert_equal sorted, @house.rooms_by_category
```
Comment (aziobrow) [inline test/house_test.rb]: nice job. as you continue to develop your skills, start to think about ways in which you can make your tests more dynamic and flexible so that they're not so easily broken if you change unimportant details (like order of rooms).
